controllers/scheduleController.js

```diff
@@ -3,8 +3,35 @@ const { getSchedule } = require('../utils')
 
 exports.getRoomSchedule = async (req, res) => {
   try {
-    return res.status(HttpStatus.OK).json(req.query.r)
-  } catch (error) {}
+    const orar = await getSchedule('./data/schedule.json')
+    if (req.query.r === undefined) {
+      return res.status(HttpStatus.OK).send(orar) // ya gettin all the rooms, man
+    }
+    const rooms = req.query.r.split(',')
+    console.log(rooms)
+    const result = {}
+    for (const an in orar) {
+      for (const zi in orar[an]) {
+        for (const entry in orar[an][zi]) {
+          for (const room in rooms) {
+            if (orar[an][zi][entry].Sala === rooms[room]) {
+              if (!Object.prototype.hasOwnProperty.call(result, an)) {
+                result[an] = {}
+              }
+              if (!Object.prototype.hasOwnProperty.call(result, zi) && !Array.isArray(result[an][zi])) {
+                result[an][zi] = []
+              }
+              result[an][zi].push(orar[an][zi][entry])
+            }
+          }
+        }
+      }
+    }
+
+    return res.status(HttpStatus.OK).json(result)
+  } catch (error) {
+    return res.send(error)
```

Comment:
we should console.error the error and return an internal error status and an object with a success field set to false and a message.